docs/quickstart.rst

```diff
@@ -129,6 +129,32 @@ certain number of old archives::
     borg prune -v $REPOSITORY --prefix '{hostname}-' \
         --keep-daily=7 --keep-weekly=4 --keep-monthly=6
 
+Pitfalls with shell variables and environment variables
+-------------------------------------------------------
+
+This applies to all environment variables you want borg to see, not just
+``BORG_PASSPHRASE``. The short explanation is: always ``export`` your variable,
+and use single quotes if you're unsure of the details of your shell's expansion
+behavior. E.g.::
+
+    export BORG_PASSPHRASE='complicated & long';
+
+This ensures the borg process that needs to access this passphrase can do so.
+Note that this is not the same thing as::
+
+  $ BORG_PASSPHRASE='complicated & long'; # missing "export"!
+  $ ./yourborgjob.sh # will never see the above $BORG_PASSPHRASE value
+
+This is because ``export`` exposes variables to subprocesses, which borg may be
+one of. More on ``export`` can be found in the "ENVIRONMENT" section of the
+bash(1) man page.
+
+.. note:: ``sudo`` does not inherit your environment, so ``sudo borg`` will not
+          have access to ``BORG_PASSPHRASE`` despite running the above example.
+
+To debug what your borg process is actually seeing, find its PID
+(``ps aux|grep borg``) and then look into ``/proc/<PID>/environ``.
```

Comment:
Do people who need this level of help really can just look into a file with zero terminated records without more hints? environ looks quite messed up for someone not into C programming and remembering that the record separators just get eaten by the terminal. I’m all for giving this hint, and keeping things short, but i’m not sure if this is more confusing like this.